src/data/set/pointwise/basic.lean

```diff
@@ -1222,6 +1222,19 @@ begin
     exact ⟨b, mem_inter (mem_smul_set.mpr ⟨a, ha, by simp [← this]⟩) hb⟩, },
 end
 
+@[to_additive]
+lemma Union_smul_eq_set_of_exists {s : set β} :
+  (⋃ (g : α), g • s) = {a | ∃ (g : α), g • a ∈ s} :=
```

Comment:
This lemma looks wrong, probably because it combines two steps together (inverting all elements of `\alpha`, commuting the image through the union). I think you rather want `Union (\la g, g\-1 \smul s) = Union (\la g, g \smul s)`.

Reply:
I think this lemma deserves to exist but I have added a commit providing `set.Union_inv_smul` and used it to golf the proof.

Reply:
Now that I think of it, I doubt either of those two new lemmas deserve to exist. The first one proves nothing about pointwise operations and the second one is three orthogonal steps together. I suspect you would better be off using `equiv.supr_congr`.

Reply:
I think both lemmas are worth having.

I admit the point is subjective but I think that the convenience they provide for such a standard situation (group actions) more than justifies the cost of having them.

I also note that `set.Union_inv_smul` is a useful `simp` lemma.

Reply:
> I suspect you would better be off using `equiv.supr_congr`.

`function.surjective.supr_congr` is a better fit here, because it gives a result that is syntactically equal to what @ocfnash wants

Reply:
Since Oliver found the lemmas useful, it's likely that they could also be useful to other people, so I'd keep them.

Reply:
> `function.surjective.supr_congr` is a better fit here

OK I've changed the proof to use `function.surjective.supr_congr` instead of `equiv.supr_congr`.